vstruct/defs/elf.py

```diff
@@ -4,6 +4,76 @@
 EI_NIDENT = 4
 EI_PADLEN = 7
 
+SHF_WRITE = 1
+SHF_ALLOC = 2
+SHF_EXECINSTR = 4
+SHF_MERGE = 16
+SHF_STRINGS = 32
+SHF_INFO_LINK = 64
+SHF_LINK_ORDER = 128
+SHF_OS_NONCONFORMING = 256
+SHF_GROUP = 512
+SHF_TLS = 1024
+SHF_ORDERED = 1073741824
+SHF_EXCLUDE = 2147483648
+
+sh_flags = {
+    SHF_WRITE:"Writable",
+    SHF_ALLOC:"Occupies memory during execution",
+    SHF_EXECINSTR:"Executable",
+    SHF_MERGE:"Might be merged",
+    SHF_STRINGS:"Contains nul-terminated strings",
+    SHF_INFO_LINK:"`sh_info' contains SHT index",
+    SHF_LINK_ORDER:"Preserve order after combining",
+    SHF_OS_NONCONFORMING:"Non-standard OS specific",
+    SHF_GROUP:"Section is member of a group.",
+    SHF_TLS:"Section hold thread-local data.",
+    SHF_ORDERED:"Special ordering",
+    SHF_EXCLUDE:"Section is excluded",
+}
+
+PT_NULL     = 0
+PT_LOAD     = 1
+PT_DYNAMIC  = 2
+PT_INTERP   = 3
+PT_NOTE     = 4
+PT_SHLIB    = 5
+PT_PHDR     = 6
+PT_TLS      = 7
+PT_NUM      = 8
+PT_LOOS     = 0x60000000
+PT_GNU_EH_FRAME  = 0x6474e550
+PT_GNU_STACK  = 0x6474e551
+PT_GNU_RELRO  = 0x6474e552
+PT_LOSUNW   = 0x6ffffffa
+PT_SUNWBSS  = 0x6ffffffa
+PT_SUNWSTACK = 0x6ffffffb
+PT_HISUNW =  0x6fffffff
+PT_HIOS   =  0x6fffffff
+PT_LOPROC =  0x70000000
+PT_HIPROC =  0x7fffffff
+
+ph_types = {
+    PT_NULL:"Program header table entry unused",
```

Comment:
if this is all duplicated from `Elf.elf_lookup`, we should have a comment at both locations.  i can't think of a good place to put them centrally.  since Elf depends on Vstruct but not viice-versa, i'm ok with them living only in vstruct... but that's terrible in its own way.  ugh.

Reply:
Yea, I tried to think of a good place to put them both centrally, but there's not a great one that doesn't cause some weird import loops, but having duplicated constants is far nicer than having weirdly slightly different structs that differ only by name.

But I will add a comment to let future us know why.